infra/blueprint-test/modules/tft/terraform.go

```diff
@@ -0,0 +1,248 @@
+	if tft.fixtureName == "" && tft.tfDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("unable to get wd :%v", err)
+		}
+		tfdir, err := discovery.ConfigDirFromCWD(cwd)
+		if err != nil {
+			t.Fatalf("unable to detect TFDir :%v", err)
+		}
+		tft.tfDir = tfdir
+	}
+	// if fixture name compute tfDir from given fixture name
+	if tft.fixtureName != "" {
+		tfModFixtureDir := getTFModuleFixtureDir(tft.fixtureName)
+		if _, err := os.Stat(tfModFixtureDir); os.IsNotExist(err) {
+			t.Fatalf("TFDir path derived from %s as %s does not exist", tft.fixtureName, tfModFixtureDir)
+		}
+		tft.tfDir = tfModFixtureDir
+	}
+	// setupPath is empty, try known setupPath
+	if tft.setupPath == "" {
+		tfModSetupDir := getTFModuleSetupDir()
+		if _, err := os.Stat(tfModSetupDir); os.IsNotExist(err) {
+			t.Logf("Setup dir %s not found, skipping loading setup outputs as fixture inputs", tfModSetupDir)
+		}
+		tft.setupPath = tfModSetupDir
+	}
+	// load TFEnvVars
+	tfEnvVars := make(map[string]string)
+	//load from setup outputs
+	if tft.setupPath != "" {
+		tft.logger.Logf(tft.t, "Loading env vars from setup %s", tft.setupPath)
+		loadTFEnvVar(tfEnvVars, tft.getTFSetupOPMap())
+	}
+	// load additional env variables
+	if tft.envVars != nil {
+		loadTFEnvVar(tfEnvVars, tft.envVars)
+	}
+	tft.envVars = tfEnvVars
+
+	tft.logger.Logf(tft.t, "Running tests TF configs in %s", tft.tfDir)
+	return tft
+}
+
+// getTFModuleFixtureDir gets TF fixture config path from a fixture name.
+func getTFModuleFixtureDir(fixture string) string {
+	return fmt.Sprintf("../../test/fixtures/%s", fixture)
+}
+
+// getTFModuleSetupDir returns well known setup path.
+func getTFModuleSetupDir() string {
+	return "../../setup"
+}
+
+// getTFOptions generates terraform.Options used by Terratest.
+func (b *TFBlueprintTest) getTFOptions() *terraform.Options {
+	return terraform.WithDefaultRetryableErrors(b.t, &terraform.Options{
+		TerraformDir: b.tfDir,
+		EnvVars:      b.envVars,
+		Vars:         b.vars,
+		Logger:       b.logger,
+	})
+}
+
+// getTFSetupOPMap computes a map of TF outputs from setup.
+// Currently only returns string outputs.
+func (b *TFBlueprintTest) getTFSetupOPMap() map[string]string {
+	o := terraform.OutputAll(b.t, &terraform.Options{TerraformDir: b.setupPath, Logger: b.logger})
+	n := make(map[string]string)
+	for k, v := range o {
+		s, ok := v.(string)
+		if !ok {
+			b.logger.Logf(b.t, "Unable to convert output %s value in setup dir to string", k)
+			continue
+		}
+		n[k] = s
+	}
+	return n
+}
+
+// GetStringOutput returns TF output for a given key as string.
+// It fails test if given key does not output a primitive.
+func (b *TFBlueprintTest) GetStringOutput(name string) string {
+	return terraform.Output(b.t, b.getTFOptions(), name)
+}
+
+// GetTFSetupOPListVal returns TF output from setup for a given key as list.
+// It fails test if given key does not output a list type.
+func (b *TFBlueprintTest) GetTFSetupOPListVal(key string) []string {
```

Comment:
What does OP stand for here?

Reply:
It stood for output but I have renamed to `GetTFSetupOutputListVal` to make it more clear.